### ISSUE TITLE
kube2iam and IRSA support

### ISSUE THREAD

**Issue by alongir** [kubeshark/kubeshark#1376]
**Current**: AWS S3 helper works only with aws key id and secret credentials. There's not support in situations that the pod is configured to use AWS at the pod level.
**Required**: 

Introduction: In situations where tools like kube2iam and IRSA are involved and pods are configured by annotations, Kubeshark's S3 helper needs to work without providing secret key and key id credentials and assume the POD is already configured.

Request #1: Helper should run without credentials and assume the pod is already configured.

To test these situations without actually setting up all of the necessary steps to support IRSA, we can use `aws config` to configure the pod and then ensure the AWS S3 helper is capable of performing without credentials.

Request #2: We then need to enable users using the helper not to pass credentials or pass empty credentials.

Request #3: Enable user to annotate the kubeshark-service account from the configuration

> Status is urgent as there's a user who can't use credentials and only kube2iam. User is waiting for resolution.

**TL;DR**
Some useful links:
https://rahullokurte.com/how-to-connect-to-s3-from-eks-using-the-iam-role-for-the-service-account

**Comment by alongir** [comment]
As an iteration that may resolve this: change helper behavior to support load default configuration incase one of the credentials is empty. There's no certainty that this will support the IRSA use-case as there's no clear correlation between AWS CLI behavior and GOLang SDK behavior.

**Comment by mertyildiran** [comment]
Implemented with:
- https://github.com/kubeshark/worker/commit/db41459a4bbde12c6004c07aca83d8dbe1b90f78
- https://github.com/kubeshark/worker/commit/daede868a04a7444f2a6620eb2faea4c6a849b98
- https://github.com/kubeshark/worker/commit/b575db43ac5d54ef8d7dd996884fd5b896ebec91
- https://github.com/kubeshark/front/commit/7d9d6a6798d32dcec98ec7a8ec7f9a17bf093ebb
- https://github.com/kubeshark/front/commit/d4be4f7b9f3a58ca5236cc7686e70020e152c401
- https://github.com/kubeshark/front/commit/93d697f36c17807d0d9a7e327daa2ac758efca17

Docs need update.

**Comment by alongir** [comment]
FYI: This does not work with EKS when the release namespace is: `default`
